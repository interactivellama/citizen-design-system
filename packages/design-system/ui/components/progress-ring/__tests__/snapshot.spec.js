// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import ProgressRing from '../';

const { matchesMarkupAndStyle } = createHelpers(__dirname);

describe('Progress Ring', () => {
  it('renders a partially drained ring', () =>
    matchesMarkupAndStyle(<ProgressRing percent={20} />));
  it('renders a partially filled ring', () =>
    matchesMarkupAndStyle(<ProgressRing percent={20} isFilling />));
  it('renders a partially filled ring with warning', () =>
    matchesMarkupAndStyle(<ProgressRing percent={20} isFilling isWarning />));
  it('renders a partially draiend ring with error', () =>
    matchesMarkupAndStyle(<ProgressRing percent={20} isExpired />));
  it('renders a complete ring', () =>
    matchesMarkupAndStyle(<ProgressRing percent={100} isComplete />));
  it('renders an empty ring', () =>
    matchesMarkupAndStyle(<ProgressRing percent={0} isFilling />));
  it('renders a filling active step ring', () =>
    matchesMarkupAndStyle(
      <ProgressRing percent={20} isActiveStep isFilling />
    ));
});
