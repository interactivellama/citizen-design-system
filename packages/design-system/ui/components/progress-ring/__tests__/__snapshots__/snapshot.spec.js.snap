// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Ring renders a complete ring 1`] = `
<div
  className="slds-progress-ring slds-progress-ring_complete"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={100}
    className="slds-progress-ring__progress"
    role="progressbar"
  >
    <svg
      viewBox="-1 -1 2 2"
    >
      <path
        className="slds-progress-ring__path"
        d="M 1 0 A 1 1 0 1 1 1 -2.4492935982947064e-16 L 0 0"
        id="slds-progress-ring-path-5"
      />
    </svg>
  </div>
  <div
    className="slds-progress-ring__content"
  />
</div>
`;

exports[`Progress Ring renders a filling active step ring 1`] = `
<div
  className="slds-progress-ring slds-progress-ring_active-step"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={20}
    className="slds-progress-ring__progress"
    role="progressbar"
  >
    <svg
      viewBox="-1 -1 2 2"
    >
      <path
        className="slds-progress-ring__path"
        d="M 1 0 A 1 1 0 0 0 0.30901699437494723 -0.9510565162951536 L 0 0"
        id="slds-progress-ring-path-7"
      />
    </svg>
  </div>
  <div
    className="slds-progress-ring__content"
  />
</div>
`;

exports[`Progress Ring renders a partially draiend ring with error 1`] = `
<div
  className="slds-progress-ring slds-progress-ring_expired"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={20}
    className="slds-progress-ring__progress"
    role="progressbar"
  >
    <svg
      viewBox="-1 -1 2 2"
    >
      <path
        className="slds-progress-ring__path"
        d="M 1 0 A 1 1 0 0 1 0.30901699437494745 0.9510565162951535 L 0 0"
        id="slds-progress-ring-path-4"
      />
    </svg>
  </div>
  <div
    className="slds-progress-ring__content"
  />
</div>
`;

exports[`Progress Ring renders a partially drained ring 1`] = `
<div
  className="slds-progress-ring"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={20}
    className="slds-progress-ring__progress"
    role="progressbar"
  >
    <svg
      viewBox="-1 -1 2 2"
    >
      <path
        className="slds-progress-ring__path"
        d="M 1 0 A 1 1 0 0 1 0.30901699437494745 0.9510565162951535 L 0 0"
        id="slds-progress-ring-path-1"
      />
    </svg>
  </div>
  <div
    className="slds-progress-ring__content"
  />
</div>
`;

exports[`Progress Ring renders a partially filled ring 1`] = `
<div
  className="slds-progress-ring"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={20}
    className="slds-progress-ring__progress"
    role="progressbar"
  >
    <svg
      viewBox="-1 -1 2 2"
    >
      <path
        className="slds-progress-ring__path"
        d="M 1 0 A 1 1 0 0 0 0.30901699437494723 -0.9510565162951536 L 0 0"
        id="slds-progress-ring-path-2"
      />
    </svg>
  </div>
  <div
    className="slds-progress-ring__content"
  />
</div>
`;

exports[`Progress Ring renders a partially filled ring with warning 1`] = `
<div
  className="slds-progress-ring slds-progress-ring_warning"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={20}
    className="slds-progress-ring__progress"
    role="progressbar"
  >
    <svg
      viewBox="-1 -1 2 2"
    >
      <path
        className="slds-progress-ring__path"
        d="M 1 0 A 1 1 0 0 0 0.30901699437494723 -0.9510565162951536 L 0 0"
        id="slds-progress-ring-path-3"
      />
    </svg>
  </div>
  <div
    className="slds-progress-ring__content"
  />
</div>
`;

exports[`Progress Ring renders an empty ring 1`] = `
<div
  className="slds-progress-ring"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={0}
    className="slds-progress-ring__progress"
    role="progressbar"
  >
    <svg
      viewBox="-1 -1 2 2"
    >
      <path
        className="slds-progress-ring__path"
        d="M 1 0 A 1 1 0 0 0 1 -2.4492935982947064e-16 L 0 0"
        id="slds-progress-ring-path-6"
      />
    </svg>
  </div>
  <div
    className="slds-progress-ring__content"
  />
</div>
`;
